.changeset/spotty-wasps-poke.md

```diff
@@ -0,0 +1,27 @@
+---
+'braid-design-system': minor
+---
+
+---
+updated:
+  - Drawer
+  - Dialog
+---
+
+**Drawer, Dialog:** Support validation blocking closure of modal
+
+To prevent a `Dialog` or `Drawer` from closing, e.g. due to validation, the `onClose` function can now return **false**.
+
+**EXAMPLE USAGE:**
+```jsx
+<Drawer
+  onClose={() => {
+    const valid = runValidation();
+    if (!valid) {
+      return false;
+    }
+
+    closeDrawer();
```

Comment:
Is this necessary? Without looking at the implementation, my expectation would be that this would let the drawer close normally.

Reply:
It is just an example. That would be the consumers controlled state function that drives the `open` prop

Reply:
Because the `open` prop is not provided in the example, I assumed it's not a controlled component.

Reply:
Yeah all good, we tend to highlight the changed prop, as using valid code with all required props can be very noisy/too verbose for a lot of components